In [1]:
import sys
from pathlib import Path

PROJECT_ROOT = Path.cwd().parent
sys.path.insert(0, str(PROJECT_ROOT))

from src.document_loader import load_all_pdfs

documents = load_all_pdfs(PROJECT_ROOT / "data" / "documents")

print("===== DOCUMENT COLLECTION OVERVIEW =====")
print("Total pages loaded:", len(documents))
print("Total source documents:", len(set(doc["source"] for doc in documents)))

print("\nDocuments:")
for source in sorted(set(doc["source"] for doc in documents)):
    pages = [doc["page"] for doc in documents if doc["source"] == source]
    print(f"- {source}: {len(pages)} pages")

Loading: 10-Q4-2024-As-Filed.pdf
  Pages loaded: 121
Loading: _10-K-2025-As-Filed.pdf
  Pages loaded: 80
Loading: _10-K-Q4-2023-As-Filed.pdf
  Pages loaded: 80
===== DOCUMENT COLLECTION OVERVIEW =====
Total pages loaded: 281
Total source documents: 3

Documents:
- 10-Q4-2024-As-Filed.pdf: 121 pages
- _10-K-2025-As-Filed.pdf: 80 pages
- _10-K-Q4-2023-As-Filed.pdf: 80 pages


In [2]:
import pandas as pd

document_summary = []

for source in sorted(set(doc["source"] for doc in documents)):
    source_documents = [
        doc for doc in documents
        if doc["source"] == source
    ]

    document_summary.append({
        "Document": source,
        "Pages": len(source_documents),
        "First Page": min(doc["page"] for doc in source_documents),
        "Last Page": max(doc["page"] for doc in source_documents)
    })

document_df = pd.DataFrame(document_summary)

document_df


,Document,Pages,First Page,Last Page
0,10-Q4-2024-As-Filed.pdf,121,1,121
1,_10-K-2025-As-Filed.pdf,80,1,80
2,_10-K-Q4-2023-As-Filed.pdf,80,1,80


In [3]:
first_document = documents[0]

print("===== TEXT EXTRACTION PREVIEW =====")
print("Source:", first_document["source"])
print("Page:", first_document["page"])
print("Extracted characters:", len(first_document["text"]))

print("\n----- Extracted Text Preview -----\n")
print(first_document["text"][:1500])

===== TEXT EXTRACTION PREVIEW =====
Source: 10-Q4-2024-As-Filed.pdf
Page: 1
Extracted characters: 1752

----- Extracted Text Preview -----

UNITED STATES
SECURITIES AND EXCHANGE COMMISSION
Washington, D.C. 20549
FORM 10-K
(Mark One)
☒    ANNUAL REPORT PURSUANT TO SECTION 13 OR 15(d) OF THE SECURITIES EXCHANGE ACT OF 1934
For the fiscal year ended September 28, 2024
or
☐    TRANSITION REPORT PURSUANT TO SECTION 13 OR 15(d) OF THE SECURITIES EXCHANGE ACT OF 1934
For the transition period from              to             .
Commission File Number: 001-36743
Apple Inc.
(Exact name of Registrant as specified in its charter)
California 94-2404110
(State or other jurisdiction
of incorporation or organization)
(I.R.S. Employer Identification No.)
One Apple Park Way
Cupertino, California 95014
(Address of principal executive offices) (Zip Code)
(408) 996-1010
(Registrant’s telephone number, including area code)
Securities registered pursuant to Section 12(b) of the Act:
Title of each class
Tradi

## Document Exploration Conclusion

The document collection contains three Apple financial filings covering fiscal years 2023, 2024, and 2025.

The ingestion pipeline successfully:
- Loaded all PDF documents using `pypdf`
- Preserved document source and page metadata
- Extracted text from individual PDF pages
- Maintained page boundaries for later source attribution

The current collection contains **281 extracted pages** across the three source documents.

These extracted documents form the knowledge base for the downstream chunking, embedding, FAISS retrieval, and RAG generation pipeline.